core/templates/dev/head/pages/skill_editor/skill_editor_navbar_directive.html

```diff
@@ -6,8 +6,9 @@
             ng-disabled="!isSkillSaveable()"
             ng-if="!isSaveInProgress()">
       <span ng-if="!isSaveInProgress()">
-        <i class="material-icons md-18 md-dark oppia-save-publish-button-icon"
-           alt="Save Skill">&#xE161;</i>
+        <i class="material-icons md-18 md-dark oppia-save-publish-button-icon" alt="Save Skill">
+           &#xE161;
```

Comment:
deindent by 1

Reply:
(@apb7 this should be a lint check?)

Reply:
Done.

Reply:
We do not process the data within the tags as of now. The parser looks into the indentation of tags and jinja templates. It's a bit difficult to keep track of the data within the tags.

Reply:
That seems odd to me. Why is it hard? I would expect it to be doable...

Reply:
Okay, so we use [HTML Parser](https://docs.python.org/2/library/htmlparser.html) for this purpose. It has a number of functions which we override -- `handle_starttag`, `handle_endtag` and `handle_data`. When a tag opens, it calls `handle_starttag`, for data, it calls, `handle_data` and when a tag closes, it calls `handle_endtag`.
The problem is, we still use jinja templates within html which is recognized by the parser just as any other text within tags and it calls the `handle_data` function. Therefore it is a bit difficult to distinguish between text with tags and other data since all are interpreted as text by the parser.

Reply:
Ah, I see, ok -- thanks for the explanation! Could we file it as an issue then, to tackle when the removal of Jinja is completed? Including the explanation you've given above in that issue would be helpful, too.

Thanks!